spec/unit/type/mount_spec.rb

```diff
@@ -530,6 +530,11 @@ def run_in_catalog(*resources)
       run_in_catalog(resource)
     end
 
+    it "should detect out of sync options" do
+      #provider.property_hash[:live_options] = "foo";
```

Comment:
This actually has caused me to realize an issue with this solution- if we're running on a non-linux platform no options will be extracted and then in the insync method they will become an empty string meaning that the options will always appear out of sync. Trying to fix this now